models/Document/Editable/Date.php

```diff
@@ -72,18 +72,17 @@ public function getDataEditmode() /** : mixed */
      */
     public function frontend()
     {
-        $format = null;
-
-        if (isset($this->config['outputFormat']) && $this->config['outputFormat']) {
-            $format = $this->config['outputFormat'];
-        } elseif (isset($this->config['format']) && $this->config['format']) {
-            $format = $this->config['format'];
-        } else {
-            $format = 'Y-m-d\TH:i:sO'; // ISO8601
-        }
-
         if ($this->date instanceof \DateTimeInterface) {
-            return $this->date->formatLocalized($format);
+            if (isset($this->config['outputFormat']) && $this->config['outputFormat']) {
+                return $this->date->formatLocalized($this->config['outputFormat']);
+            } else {
+                if (isset($this->config['format']) && $this->config['format']) {
+                    $format = $this->config['format'];
+                } else {
+                    $format = 'Y-m-d\TH:i:sO'; // ISO8601
```

Comment:
Why not use [a constant](https://www.php.net/manual/en/class.datetimeinterface.php#datetimeinterface.constants.types) here?

Note that you currently use the value of the `DateTimeInterface::ISO8601` constant, which is **not** compatible with `ISO-8601`!

`DateTimeInterface::ATOM` is the constant to use when `ISO-8601` is desired.

```suggestion
                    $format = DateTimeInterface::ATOM; // ISO8601
```

Reply:
Yes, that will be covered in #13633, funnily this bug was *discovered* while working on that